airflow/providers/cncf/kubernetes/backcompat/backwards_compat_converters.py

```diff
@@ -21,18 +21,16 @@
 from kubernetes.client import ApiClient, models as k8s
 
 from airflow.exceptions import AirflowException
-from airflow.providers.cncf.kubernetes.backcompat.pod import Port, Resources
-from airflow.providers.cncf.kubernetes.backcompat.pod_runtime_info_env import PodRuntimeInfoEnv
 
 
-def _convert_kube_model_object(obj, old_class, new_class):
+def _convert_kube_model_object(obj, new_class):
     convert_op = getattr(obj, "to_k8s_client_obj", None)
     if callable(convert_op):
         return obj.to_k8s_client_obj()
```

Comment:
@jedcunningham, do you think we should add deprecation warning here in addition to [here](https://github.com/apache/airflow/pull/19726/files#diff-fe570c8d3e660e4fca5e3d1f4be8f3c075d30c37d41d2f66f90d5bfbde964f33R76)?

Users who import the backcompat modules will get warnings when they do so; but maybe we should _also_ signal here that this logic to convert will be removed.

Reply:
PS: Isn't it implied by definition of "Deprecation"?

Reply:
I don't think we need to show 2 deprecation warnings, and I agree with @dimon222 that it's implied the backcompat conversion will eventually be removed.

Reply:
> PS: Isn't it implied by definition of "Deprecation"?

Removal is implied by the word deprecation, yes.   However, KPO is not making any deprecation warning (which is my concern) yet the KPO _will eventually_ remove this logic.  I think 2 deprecations warnings is the lesser of 2 evils here (one for "backcompat is gonna be removed" the other for "we aren't gonna convert forever"), but not a hill to die on and i don't stand in the way.

Further, the logic [here](https://github.com/apache/airflow/pull/19726/files#diff-fe570c8d3e660e4fca5e3d1f4be8f3c075d30c37d41d2f66f90d5bfbde964f33R76) does  not even assume a backcompat class -- it is  logic to accept kwargs from a dict.  So in that case the user won't have a deprecation warning at all.  Again no dying on hills for me here either and i defer to @jedcunningham

Reply:
@jedcunningham got any better suggestion?

Reply:
At least in my eyes, "backcompat" and "convert" is the same thing 🤷‍♂️. I don't think we need to warn about both, but the nuance may be more important for end users. 

In cases where there users aren't using the "old" classes directly, like your `resources` example (which will
 actually get a deprecation warning in #20031), I think we should swallow the deprecation warning and emit a more helpful one for the user.

For example, for users using dict resources, instead of saying:
```
This module is deprecated. Please use `kubernetes.client.models.V1ResourceRequirements` and `kubernetes.client.models.V1ContainerPort`."
```

say (or something like it):

```
Setting resources as a dict is deprecated. Please use `kubernetes.client.models.V1ResourceRequirements`
```

`env_vars` also has this issue, not sure if there are others though.

Either way, I think the new deprecation warning stuff should be handled in #20031.

Reply:
I guess thats a problem of the fact that Resources and Port are combined together under not straightforward named "pod.py", otherwise the filename itself should describe the exact concerning class (ala resources.py and port.py). But you're right, its better be handled separately in #20031. So, should we merge current PR then as is?